extensions/agent/client/src/dialogs/createStepDialog.ts

```diff
@@ -0,0 +1,489 @@
+				'C:\\Program Files\\Microsoft SQL Server\\MSSQL14.MSSQLSERVER\\MSSQL\\Backup',
+				['*'] , false).then(result => {
+				if (result) {
+					console.log(result);
+					Promise.resolve(result);
+				} else {
+					Promise.reject(false);
+				}
+			});
+		});
+		this.outputFileNameBox = view.modelBuilder.inputBox()
+			.withProperties({
+				width: '100px',
+				inputType: 'text'
+			}).component();
+		let outputViewButton = view.modelBuilder.button()
+			.withProperties({
+				width: '50px',
+				label: 'View'
+			}).component();
+		outputViewButton.enabled = false;
+		let outputButtonContainer = view.modelBuilder.flexContainer()
+			.withLayout({
+				flexFlow: 'row',
+				textAlign: 'right',
+				width: 120
+			}).withItems([this.outputFileBrowserButton, outputViewButton], { flex: '1 1 50%'}).component();
+		let outputFlexBox = view.modelBuilder.flexContainer()
+			.withLayout({
+				flexFlow: 'row',
+				width: 350
+			}).withItems([this.outputFileNameBox, outputButtonContainer], {
+				flex: '1 1 50%'
+			}).component();
+		this.appendToExistingFileCheckbox = view.modelBuilder.checkBox()
+			.withProperties({
+				label: 'Append output to existing file'
+			}).component();
+		this.appendToExistingFileCheckbox.enabled = false;
+		this.outputFileNameBox.onTextChanged((input) => {
+			if (input !== '') {
+				this.appendToExistingFileCheckbox.enabled = true;
+			} else {
+				this.appendToExistingFileCheckbox.enabled = false;
+			}
+		});
+		let outputFileForm = view.modelBuilder.formContainer()
+			.withFormItems([{
+					component: outputFlexBox,
+					title: 'Output file'
+				}, {
+					component: this.appendToExistingFileCheckbox,
+					title: ''
+				}], { horizontal: true, componentWidth: 200}).component();
+		return outputFileForm;
+	}
+
+	private async execute() {
+		this.model.jobId = this.jobId;
+		this.model.server = this.server;
+		this.model.stepName = this.nameTextBox.value;
+		this.model.subSystem = this.typeDropdown.value as string;
+		this.model.databaseName = this.databaseDropdown.value as string;
+		this.model.script = this.commandTextBox.value;
+		this.model.successAction = this.successActionDropdown.value as string;
+		this.model.retryAttempts = +this.retryAttemptsBox.value;
+		this.model.retryInterval = +this.retryIntervalBox.value;
+		this.model.failureAction = this.failureActionDropdown.value as string;
+		this.model.outputFileName = this.outputFileNameBox.value;
+		await this.model.save();
+	}
+
+	private openFileBrowserDialog(rootNode, selectedNode) {
+	}
+
+	private onFileBrowserOpened(handle: number, fileBrowserOpenedParams: sqlops.FileBrowserOpenedParams) {
+		if (fileBrowserOpenedParams.succeeded === true
+			&& fileBrowserOpenedParams.fileTree
+			&& fileBrowserOpenedParams.fileTree.rootNode
+			&& fileBrowserOpenedParams.fileTree.selectedNode) {
+				this.openFileBrowserDialog(fileBrowserOpenedParams.fileTree.rootNode, fileBrowserOpenedParams.fileTree.selectedNode);
+			}
+		console.log('no response');
```

Comment:
this looks like debug code that should be deleted.  Also, below `return isn't needed.